doc/source/whatsnew/v0.21.0.txt

```diff
@@ -296,6 +296,7 @@ Groupby/Resample/Rolling
 - Bug in :func:`infer_freq` causing indices with 2-day gaps during the working week to be wrongly inferred as business daily (:issue:`16624`)
 - Bug in ``.rolling(...).quantile()`` which incorrectly used different defaults than :func:`Series.quantile()` and :func:`DataFrame.quantile()` (:issue:`9413`, :issue:`16211`)
 - Bug in ``groupby.transform()`` that would coerce boolean dtypes back to float (:issue:`16875`)
+- Bug in ``DataFrame.groupby`` when called with index or index + column key numbers equal to the axis length of the groupby (:issue:`16859`)
```

Comment:
Your description underneath your test I think is clearer about the bug than this description is.  Try incorporating some of your test description into here.